cmd/telegraf/telegraf.go

```diff
@@ -152,32 +152,39 @@ func runAgent(ctx context.Context,
 			c.Agent.Interval.Duration)
 	}
 
-	ag, err := agent.NewAgent(c)
+	if runningAgent != nil && runningAgent.Config != nil {
+		for idx := range runningAgent.Config.Outputs {
+			log.Printf("I! Closing running output: %s", runningAgent.Config.Outputs[idx].LogName())
+			runningAgent.Config.Outputs[idx].Close()
+		}
+	}
+
+	runningAgent, err = agent.NewAgent(c)
```

Comment:
Please also set `runningAgent = nil` in case of an error. Otherwise we might run into trouble if the agent address is a leftover from a previous run...